doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,255 @@
-training code. If you need access to more fine grained APIs, use
-the :ref:`native integrations directly <train-experiment-tracking-native>`.
+    .. tab:: wandb
 
-Example: Logging to MLflow and TensorBoard
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        .. literalinclude:: ../doc_code/lightning_experiment_tracking.py
+            :language: python
+            :start-after: __lightning_experiment_tracking_wandb_start__
+            :end-before: __lightning_experiment_tracking_wandb_end__
 
-**Step 1: Install the necessary packages**
+    .. tab:: comet
 
-.. code-block:: bash
+        .. literalinclude:: ../doc_code/lightning_experiment_tracking.py
+            :language: python
+            :start-after: __lightning_experiment_tracking_comet_start__
+            :end-before: __lightning_experiment_tracking_comet_end__
 
-    $ pip install mlflow
-    $ pip install tensorboardX
+    .. tab:: mlflow
 
-**Step 2: Run the following training script**
+        .. literalinclude:: ../doc_code/lightning_experiment_tracking.py
+            :language: python
+            :start-after: __lightning_experiment_tracking_mlflow_start__
+            :end-before: __lightning_experiment_tracking_mlflow_end__
 
-.. literalinclude:: /../../python/ray/train/examples/mlflow_simple_example.py
-   :language: python
+    .. tab:: tensorboard
+        
+        .. literalinclude:: ../doc_code/lightning_experiment_tracking.py
+            :language: python
+            :start-after: __lightning_experiment_tracking_tensorboard_start__
+            :end-before: __lightning_experiment_tracking_tensorboard_end__
+        
+.. tip::
+    
+    When performing **fault-tolerant training** with auto-restoration, be sure 
+    to specify a unique ID for the Loggers, so that the new workers report to
+    the same run after restoration.
+
+    For example:
+    
+    - `WandbLogger(id=UNIQUE_ID)`
+    - `CometLogger(experiment_key=UNIQUE_ID)`
+    - `MLFlowLogger(run_id=UNIQUE_ID)`
+
+Legacy APIs
+===========
+
+Should I use ``setup_wandb`` and ``setup_mlflow``?
+--------------------------------------------------
+
+``setup_wandb`` and ``setup_mlflow`` are just convenient wrappers on top of
+Weights&Biases and MLflow's native APIs. To spare you from learning yet another
+set of APIs, we recommend you to continue using the native experiment tracking 
+directly.
+Just remember to guard with ``rank==0`` to avoid duplications whenever applicable.
+
+Words on ``MLflowLoggerCallback`` and ``WandbLoggerCallback``
+-------------------------------------------------------------
+
+More seasoned Ray library users may wonder “Wait, what about `MLflowLoggerCallback
+` and `WandbLoggerCallback`?” 
+These APIs rely on reporting results to Ray Train and then log to experiment training 
+frameworks. The usage scenario is not as versatile and requires some changes to the 
+tracking logic. As a result, we advise users against using these callbacks when using any
+DataParallelTrainer.
```

Comment:
Would personally prefer not to include these here, as it'll be more confusing for new users.

Reply:
Can we say something to guard against confusion? 
My worry about moving it entirely is the existing users. They are going to ask about these questions for sure.

Reply:
removed entirely